docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+   workers so that all workers decide to preempt or continue as a unit.
+
+..
+   tip Launchers
+
+   Typically, you do not have to write your own launcher.
+   Determined provides launchers for Horovod, torch.distributed, and
+   DeepSpeed. For more information about launcher options, visit
+   :doc:`Submit Experiment </training/submit-experiment>`.
+
+To perform distributed training with the Core API, you’ll need to use the appropriate distributed
+training library *before* creating a Determined :doc:`Distributed Context
+</reference/reference-training/training/api-core-reference>`.
+
+In this example, we’ll be using PyTorch’s DistributedDataParallel. We’ll also need to make specific
+changes to our configuration experiment (YAML) file.
+
+In this step, we’ll run our experiment using the model_def_distributed.py script and its
+accompanying distributed.yaml experiment configuration file.
+
+Step 5.1: Edit Your Experiment Configuration (YAML) File
+========================================================
+
+Edit your experiment configuration (YAML) file to point to a launch script:
+
+.. code:: yaml
+
+   entrypoint: >-
+      python3 -m determined.launch.torch_distributed
+      python3 model_def_distributed.py
+
+and, set ``slots_per_trial`` (under ``resources``) to the number of GPUs you want to distribute the
+training across:
+
+.. code:: yaml
+
+   resources:
+     slots_per_trial: 4
+
+.. image:: /assets/images/distributed-yaml.png
+
+Step 5.2: Modify Your Training Script
+=====================================
+
+Add a few more imports to your training script:
+
+.. code::
+
+   # NEW: Import torch distributed libraries.
+   import torch.distributed as dist
+   from torch.utils.data.distributed import DistributedSampler
+   from torch.nn.parallel import DistributedDataParallel as DDP
+
+.. image:: /assets/images/model-def-distributed-script.png
+
+Initialize a process group and a Determined distributed context using ``from_torch_distributed``:
+
+.. code::
+
+   if __name__ == "__main__":
+       # NEW: Initialize process group using torch.
+       dist.init_process_group("nccl")
+
+       # NEW: Initialize distributed context using from_torch_distributed
+       # (obtains info such as rank, size, etc. from default torch environment variables).
+       distributed = det.core.DistributedContext.from_torch_distributed()
+       with det.core.init(distributed=distributed) as core_context:
+           main(core_context)
+
+In ``main``, set your device to the one on index ``local_rank``. This is a best practice even if you
+only have a single GPU-per-node setup:
+
+..
+   note:
+
+   Refer to line 187 in ``model_def_distributed.py``
+
+.. code::
+
+   # NEW: Change selected device to the one with index of local_rank.
+   device = torch.device(core_context.distributed.local_rank)
+
+Shard the data into ``num_replicas`` non-overlapping parts. ``num_replicas`` is equal to
+\`core_context.distributed.size, or the number of slots):
```

Comment:
what's going on with this backslash backtick?

Reply:
resolved

`Shard the data into ``num_replicas`` non-overlapping parts. ``num_replicas`` is equal to
``core_context.distributed.size``, or the number of slots:`